engine/src/savegame.cpp

```diff
@@ -174,6 +180,99 @@ void SaveFileCopy( const char *src, const char *dst )
     }
 }
 
+class Utf8Checker {
+public:
+    bool validUtf8(const std::vector<BYTE> &data) {
+        for (long unsigned int i = 0; i < data.size(); ++ i) {
+            // 0xxxxxxx
+            int bit1 = (data[i] >> 7) & 1;
```

Comment:
Might want to make these `bitN` variables `unsigned int`s

Reply:
You're welcome to modify anything as you see fit, It's mostly code from [here](https://helloacm.com/how-to-validate-utf-8-encoding-the-simple-utf-8-validation-algorithm/), practically untouched.